### ISSUE TITLE
`devServerPublicPathRoute` omitted from `ConfigOptions`

### ISSUE THREAD

**Issue by trbagley-gresham** [cypress-io/cypress#29633]
### Current behavior

When setting `devServerPublicPathRoute` in `defineConfig`, TypeScript complains that the property isn't valid despite it being [documented](https://docs.cypress.io/guides/references/configuration#component).

### Desired behavior

TypeScript should accept `devServerPublicPathRoute` as a valid property of `component` in `defineConfig`.

### Test code to reproduce

Set `devServerPublicPathRoute` in _cypress.config.ts_:
```js
export default defineConfig({
  component: {
    devServer: {
      framework: 'react',
      bundler: 'vite'
    }
  },
  devServerPublicPathRoute: '',
});
```

### Cypress Version

13.11.0

### Node version

20.11.1

### Operating System

macOS 14.5

**Comment by AtofStryker** [comment]
Hi @trbagley-gresham. Thank you for opening an issue. I think this may have been oversight on my end when publicly exposing the option since right now its configured to be an internal option https://github.com/cypress-io/cypress/blob/develop/packages/config/src/options.ts#L530

**Comment by jennifer-shehane** [comment]
@AtofStryker Shouldn't this be within the `component` object of the config? The example shows it as being set outside of the `component` object.

**Comment by trbagley-gresham** [comment]
@jennifer-shehane That's correct, my mistake, although the same happens when `devServerPublicPathRoute` is placed within the `component` object. TypeScript `tsc` doesn't appear to report any issues though.
